cli/src/main.rs

```diff
@@ -0,0 +1,262 @@
+use bson::{doc, Document};
+use clap::Parser;
+use mongodb::sync::{Client, Collection};
+use mongosql::{catalog::Catalog, json_schema::Schema, Namespace};
+use std::collections::{BTreeMap, BTreeSet};
+
+const SQL_SCHEMAS_COLLECTION: &str = "__sql_schemas";
+
+#[derive(Debug)]
+struct CliError(String);
+
+impl std::fmt::Display for CliError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl<T> From<T> for CliError
+where
+    T: std::error::Error,
+{
+    fn from(e: T) -> Self {
+        CliError(e.to_string())
+    }
+}
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about=None)]
+struct Cli {
+    #[arg(
+        short,
+        long,
+        help = "The current database where collections in the query are assumed to live (cross database queries are not supported), default = test"
+    )]
+    db: Option<String>,
+    #[arg(index = 1, help = "The query to translate")]
+    query: String,
+    #[arg(
+        short,
+        long,
+        help = "translation, automatically true if result not set"
+    )]
+    translation: bool,
+    #[arg(
+        short,
+        long,
+        help = "Display the result of this query when applied against the mongodb uri"
+    )]
+    result: bool,
```

Comment:
I wasn't too sure what this would do until I ran the executable.  I'm wondering if this would be clearer if named `execute` or `run` with the following `help`:
```
        help = "Execute the query against the MongoDB URI and display the results"
```